mbed_flasher/flashers/FlasherMbed.py

```diff
@@ -303,3 +312,21 @@ def verify_flash_success(self, new_target, target, tail):
 
         self.logger.debug("ready")
         return EXIT_CODE_SUCCESS
+
+    def _check_mount_point_disappear(self, mount_point):
+        # check if mount point disappear after file copied into device
+        mount_point_previous_mounted = None
+
+        for _ in range(self.MOUNT_POINT_DISAPPEAR_TIMEOUT):
+            if os.path.ismount(mount_point):
+                mount_point_previous_mounted = True
+
+            if not os.path.ismount(mount_point) and mount_point_previous_mounted is True:
+                self.logger.info("mount point disappears now after file copied")
+                return
+
+            sleep(1)
```

Comment:
what about if we just decrease delay inside loop, e.g.  `sleep(0.5)`. At least then probability would be significantly smaller that we doesn't recognise mount point disappearing.. or maybe that could be even removed like you mention already.. hmm. This need to be investigate a bit more before merging.

Reply:
Still `sleep(0.5)` does not guarantee it is 0.5 seconds.

Reply:
I know. Do you have any better idea?

Reply:
Is this necessary to catch mount point disappearance?

Reply:
otherwise we might check state in wrong phase and make decision too early. Originally there was 4s hard codes delay when mount point disappeared 99% of cases. But because it is not robust enough we need better approach - target in this PR. Better approach would be to “polling” `remount_count` from details.txt but because it doesn’t exist in all boards it doesn’t solve fully this challenge. When it does we should use it, but what about those other boards? Can we drop support? maybe disappear check loop could be used for rest of boards.. If it would be easy to get trigger directly from OS (all of them) about when mount point disappears based on target id it probably would be most generic option here.. Drag’n’drop flashing approach is a bit complicated from  automation point of view.

Reply:
What state and what decision? I thought it is enough to see `.bin` disappear to decide that remount happened? What about flashing failure?

I described exactly the same corner case to @YifeiZuo01 than you did above, but seems like there was either some logical error in my explanation or I did not know enough about mbed-flasher. I think the corner case is still valid.

Reply:
That might be problematic to check `.bin` disappear.
1.. currectly we do flash `.hex` and others, even if we enumrate all the ones we are currently using, there is still limit to users what kind of file mbed-flasher supports.
2. for the logic to check `.bin` or `.hex` disappear,  the appearance of `.bin` needed to be checked, cause non-exists of `.bin` file, doesn't necessaryily mean it's been flashed successfully. 
Tracking the changes of appearance of binary, which probably also need sleep().